docs: Review Auth page (#2867)
vectorizedio/redpanda: docs/www/acls.md

On `rpk`'s side, you can use the `--tls-key`, `--tls-cert` and `--tls-truststore` flags to have it authenticate and establish a TLS connection:

```bash
 rpk topic create test-topic \
--tls-key <path to PEM-formatted key file> \
--tls-cert <path to PEM-formatted cert file> \
--tls-truststore <path to PEM-formatted truststore file>
```

```bash
Created topic 'test-topic'.
```

You may check its config with:

```bash
rpk topic describe 'test-topic'
```

**_Note_** - `rpk` defaults to connecting to `localhost:9092`. If you're connecting to a remote broker, you will need to pass `--brokers <node IP>:<kafka API port>`

For an in-depth guide about enabling TLS and mutual TLS in redpanda check out [our guide on the subject](https://vectorized.io/blog/tls-config/).

### SASL/ SCRAM

> Redpanda also supports client authentication via SASL/SCRAM (that is, the Simple Authentication and Security Layer protocol, using the Salted Challenge Response Authentication Mechanism), which is based on usernames & passwords.

Author: I thought about it, but linking a Wikipedia article doesn't seem right. A RFC is overkill too (most people don't bother with it). I'm not sure what's best here but I don't like this part too.

> Redpanda also supports client authentication via SASL/SCRAM (that is, the Simple Authentication and Security Layer protocol, using the Salted Challenge Response Authentication Mechanism), which is based on usernames & passwords.

Reviewer: Referring to Wikipedia is pretty common for topics like this.

To enable it, set `redpanda.enable_sasl` to `true` in the configuration file, as well as list at least one "superuser" which after being created will have permissions for all operations on the clusters.

You config must look something like this:

```yaml
redpanda
  
  enable_sasl: true
  superusers:
    - username: admin
  # The rest of the config...
```

<tabs>
  <tab id="Local Redpanda">
  You can find the configuration file here

  ```bash
  /etc/redpanda/redpanda.yaml
  ```

> After you change it's value, restart Redpanda service for changes to be in effect. 

Reviewer: Correct grammar. "it's value" sounds like you are changing the value of one parameter where here you might be changing values of multiple parameters? Maybe use, "After you change the config...".

```suggestion
After you change it's value, restart Redpanda service for changes to take effect. 
```

  </tab>
  <tab id="Docker Container">
  To access the files inside the container, first you have to enter the container.

  You can do that by running:

```bash
  docker exec -it <name-of-container> bash
```

Change <name-of-container> for your container and execute the command. 

> Then go to the same directory as a local redpanda config:

Reviewer: Why do you need to go to the redpanda config dir?

```bash
  /etc/redpanda/redpanda.yaml
```

After you finish it, restart the container for changes to be in effect. 

  </tab>
  <tab id="Kubernetes">
  For Kubernetes, things are a little different. You can change your configuration file with the YAML for the cluster.

>   If you got our [one-cluster sample](https://github.com/vectorizedio/redpanda/blob/dev/src/go/k8s/config/samples/one_node_cluster.yaml) you should add these values to enable different flags.

Reviewer: - Correct grammar
- "should" indicates a possibility whereas this is the required way to set SASL in K8s.
- Specify the fields to change

```suggestion
  If you're using our [one-cluster sample](https://github.com/vectorizedio/redpanda/blob/dev/src/go/k8s/config/samples/one_node_cluster.yaml), specify the `redpanda.enable_sasl` and `superuser` values in the cluster spec YAML file.
```

  For example: 

```yaml
apiVersion: redpanda.vectorized.io/v1alpha1
kind: Cluster
metadata:
  name: external-connectivity
spec:
  image: "vectorized/redpanda"
  version: "latest"
  replicas: 3
  enableSasl: true
  superUsers:
    - username: admin
  resources:
    requests:
      cpu: 1
      memory: 2Gi
    limits:
      cpu: 1
      memory: 2Gi
  configuration:
    rpcServer:
      port: 33145
    kafkaApi:
     - port: 9092
     - external:
         enabled: true
    pandaproxyApi:
     - port: 8082
     - external:
         enabled: true
    adminApi:
    - port: 9644
    developerMode: true
```

> **_Note_** - The attributes in K8s YAML are in camelCase instead of snake_case used in local redpanda.yaml

Reviewer: `>` indents the note

```suggestion
> **_Note_** - The attributes in Kubernetes YAML are in camelCase instead of snake_case used in the local redpanda.yaml
```

> Remember, that after you change the config file you **must restart** the pods for changes to be in effect.

Reviewer: I might emphasize the action "restart the nodes" or just "must".

You can check if your spec is correct by running: 

```
kubectl get clusters external-connectivity -o=jsonpath='{.spec}' 
```

  </tab>
</tabs>

Then, set a password for the user by running the following command. Replace `<password>` for a password of your choice. 

```bash
rpk acl user create \
--new-username admin \
--new-password <password> \
--api-urls localhost:9644
```

The response should be:

```
Created user 'admin'
```

> **_Note_** - `rpk acl` commands are defaulted to connect to `localhost:9644`. If you have different configuration, you will need to pass `api-urls <node IP>:<admin API port>`

Reviewer: - `>` indents the note
- Avoid "will"

```suggestion
> **_Note_** - `rpk acl` commands are defaulted to connect to `localhost:9644`. If you use a different port, pass `api-urls <node IP>:<admin API port>`
```

Author: I really mean configuration here, because if the guy sets a domain for example, he will need to change both node IP and admin port.

Reviewer: Maybe "IP address, URL, or port..."

  0                   1               [1]        [1]               0               
```

> **_Note_** - If you still have the TLS config from the previous section, you'll also need to pass the TLS flags.

Reviewer: `>` indents the note

```suggestion
> **_Note_** - If you still have the TLS config from the previous section, you'll also need to pass the TLS flags.
```

> Notice that this command uses Kafka API, so we're using `--brokers <node IP>:<kafka API port>` pattern here.

Reviewer: Correct grammar

```suggestion
Notice that this command uses the Kafka API, so we're using the `--brokers <node IP>:<kafka API port>` pattern here.
```

#### Managing users

While having a superuser is a must to get started, it's not really a good idea to either share the superuser's credentials everywhere or to make everyone a superuser.

You can create, delete and list users with `rpk acl user`.

> Remember that since this is a `rpk acl` command the default value of ``--api-urls`` is  `localhost:9644`. If you have a different configuration, you will need to pass `api-urls <node IP>:<admin API port>`.

Reviewer: Simplify sentence.

```suggestion
The default value of `--api-urls` in the acl command is `localhost:9644`. If you use a different port, pass `api-urls <node IP>:<admin API port>`.
```

Author: I really mean configuration here, because if the guy sets a domain for example, he will need to change both node IP and admin port.
But the simplified version is indeed better.

Author: I've rewrote this sentence

Replace `<password>` for a password of your choice. 

**Creating a user**

<tabs>

  <tab id="Local Redpanda">

```bash
rpk acl user create \
--new-username Jack \
--new-password <password> \
--api-urls localhost:9644
```

```bash 
Created user 'Jack'
```

**Deleting a user**

```bash
rpk acl user delete \
--delete-username Jack \
--api-urls localhost:9644
```

```bash
Deleted user 'Jack'
```

**Listing users**

```bash
rpk acl user list \
--api-urls localhost:9644
```

```bash
USERNAME                        
                                
Michael                         
Jim                             
Pam                             
Dwight                          
Kelly
```

  </tab>

  <tab id="Kubernetes">

> Remember to change <your-cluster> to your own values.

Reviewer: Code format

```suggestion
Remember to change `<your-cluster>` to your own values.
```
